src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1509,57 +1509,51 @@ async Task EnsureResourceDeletedAsync<T>(string resourceName) where T : CustomRe
         {
             _logger.LogDebug("Ensuring '{ResourceName}' is deleted.", resourceName);
 
-            var resourceNotFound = false;
-            string? uid = null;
-            try
-            {
-                var r = await _kubernetesService.DeleteAsync<T>(resourceName, cancellationToken: cancellationToken).ConfigureAwait(false);
-                uid = r.Uid();
-
-                _logger.LogDebug("Delete request for '{ResourceName}' successfully completed. Resource to delete has UID '{Uid}'.", resourceName, uid);
-            }
-            catch (HttpOperationException ex) when (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+            var result = await DeleteResourceRetryPipeline.ExecuteAsync<bool, string>(async (resourceName, attemptCancellationToken) =>
```

Comment:
Maybe add a comment here about why both the delete and get are inside the retry.

Reply:
Done.